macros/src/lib.rs

```diff
@@ -68,6 +97,61 @@ struct InputData {
 
     #[darling(default)]
     strict: bool,
+
+    #[darling(default)]
+    rename_all: Option<String>,
+}
+
+#[derive(Debug, FromDeriveInput)]
+#[darling(attributes(try_from_field), supports(enum_unit))]
+struct TryFromFieldInputData {
+    ident: syn::Ident,
+    data: darling::ast::Data<FieldEnumData, ()>,
+
+    #[darling(default)]
+    rename_all: Option<String>,
+}
+
+/// Derive `TryFromField` for arbitrary unit-enums.
+#[proc_macro_error]
+#[proc_macro_derive(TryFromField, attributes(try_from_field, field))]
+pub fn try_from_field(input: TokenStream) -> TokenStream {
```

Comment:
How would you feel about splitting the implementation into multiple files? Personally I would split it into a try_from_field and try_from_multipart files so it's easier to read, but your call, this looks good too.

Reply:
Yep, did this in the latest commit. Is that structure fine?